### ISSUE TITLE
vstack not equivalent to concatenate(x, axis=0) (migrated from Trac #1446)

### ISSUE THREAD

**Issue by thouis** [thouis/numpy-trac-migration#2998]
Original ticket http://projects.scipy.org/numpy/ticket/1446
Reported 2010-04-04 by atmention:mforbes, assigned to unknown.

The documentation states that {{{vstack(tup) is equivalent to concatenate(tup, axis=0)}}}.  This fails for 1D arrays because of the promotion 1D->2D.

```
>>> tup = [np.zeros(1), np.zeros(5)]
>>> np.concatenate(tup, axis=0)
array([ 0.,  0.,  0.,  0.,  0.,  0.])
>>> vstack(tup)
Traceback (most recent call last):
   ...
ValueError: array dimensions must agree except for d_0
```

If the current behaviour of automatic promotion is desired, then this is a documentation bug and a bug with the exception message (all the dimensions do agree except for d_0 in this case -- they don't exist!)  I will add a note about this on the documentation page too.

**Comment by thouis** [comment]
Comment in Trac by atmention:rgommers, 2011-03-30

Fixed docs in commit:d8b1b5a, thanks.

**Comment by thouis** [comment]
Comment in Trac by atmention:charris, 2012-04-26
